File: python/restart/deo.c
#include <stdio.h>
#include <string.h>

void comp_str(char argone[100], char argtwo[100]);

int main(void)
{
	comp_str("deobaba", "baba");
	comp_str("deobaba", "bbaa");

	return (0);
}

void comp_str(char *argone, char *argtwo)
{
    int i = 0;

    if (strstr(argone, argtwo) != NULL)
		printf("true\n");
	else
		printf("false\n");
}

File: python/restart/deo.py
def comp_str(argone, argtwo):
    argone_rev = list(argone[::-1])
    argtwo_rev = list(argtwo[::-1])

    for char in argone_rev:
        if argtwo_rev and char == argtwo_rev[0]:
            argtwo_rev.pop(0)
        else:
            break

    if argtwo_rev:
        print(False)
    else:
        print(True)


comp_str("deobaba", "baba")
comp_str("deobaba", "bbaa")
